tests/virtualization/syscontainer_image_test.pm

```diff
@@ -0,0 +1,128 @@
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: system containers images test setup, installing libvirt-lxc
+# Maintainer: Cédric Bosdonnat <cbosdonnat@suse.de>
+
+use base "basetest";
+use testapi;
+use strict;
+
+sub run() {
+    select_console 'root-console';
+
+    # Check the repositories on the host first
+    assert_script_run("zypper lr");
+
+    my $url = get_var('SYSCONTAINER_IMAGE_URL');
+    my $rootfs = '/tmp/rootfs';
+
+    # Create XML file
+    my $scccredentials_mount = << "EOF";
+    <filesystem type='mount' accessmode='passthrough'>
+      <source dir='/etc/zypp/credentials.d/SCCcredentials'/>
+      <target dir='/etc/zypp/credentials.d/SCCcredentials'/>
+    </filesystem>
+EOF
+    if (!get_var('SCC_REGCODE'))
+    {
+	$scccredentials_mount = '';
+    }
+
+    my $xml = << "EOF";
+<domain type='lxc'>
+  <name>guest</name>
+  <memory unit='MiB'>256</memory>
+  <currentMemory unit='MiB'>256</currentMemory>
+  <vcpu placement='static'>1</vcpu>
+  <resource>
+    <partition>/machine</partition>
+  </resource>
+  <os>
+    <type arch='x86_64'>exe</type>
+    <init>/sbin/init</init>
+  </os>
+  <clock offset='utc'/>
+  <on_poweroff>destroy</on_poweroff>
+  <on_reboot>restart</on_reboot>
+  <on_crash>restart</on_crash>
+  <devices>
+    <filesystem type='mount' accessmode='passthrough'>
+      <source dir='$rootfs'/>
+      <target dir='/'/>
+    </filesystem>
+$scccredentials_mount
+    <filesystem type='mount' accessmode='passthrough'>
+      <source dir='/etc/resolv.conf'/>
+      <target dir='/etc/resolv.conf'/>
+    </filesystem>
+    <interface type='network'>
+      <source network='default'/>
+      <ip address='192.168.122.5' family='ipv4' prefix='24'/>
+      <route address='0.0.0.0' prefix='0' gateway='192.168.122.1'/>
+    </interface>
+    <console type='pty'>
+      <target type='lxc' port='0'/>
+    </console>
+    <hostdev mode='capabilities' type='misc'>
+      <source>
+        <char>/dev/ttyS0</char>
+      </source>
+    </hostdev>
+  </devices>
+</domain>
+EOF
+    assert_script_run("echo \"$xml\" >test.xml");
+
+    # Setup root filesystem
+
+    #   - Unpack the image
+    assert_script_run("curl -L -o syscontainer-image.tar.bz2 $url");
+    assert_script_run("mkdir $rootfs");
+    assert_script_run("tar xf syscontainer-image.tar.bz2 -C $rootfs");
+
+    #   - Set root password
+    assert_script_run("echo 'root:test' | chpasswd --root $rootfs");
+
+    # Create the container
+    script_run("virsh -c lxc:/// create --console test.xml", 0);
+
+    # Check that the container actually started
+    assert_screen('syscontainer-login');
+
+    # Tests on the container
+
+    # Test login on the container
+    type_string("root\n");
+    type_password("test\n");
+
+    # Sort by command since we can't check the PIDs:
+    # this makes the needle more robust
+    script_run("ps -e --sort ucmd");
```

Comment:
better use `assert_script_run` or, because you check the output anyway with an assert_screen -> `script_run …, 0`